### ISSUE TITLE
Don't refresh the page when there are unsaved changes!

### ISSUE THREAD

**Issue by Revadike** [DestinyItemManager/DIM#10761]
### DIM Version

Version 8.42.0.3621 (beta), built on 15/10/2024, 23:51:21

### Browser Details

Firefox 131.0.2 (64-bit)

### OS Details

Windows 11

### Describe the bug

I was making a build and I accidentally clicked on another tab that had DIM open, and it refreshed both tabs, discarding my whole build...
I'd like to request 2 changes, one that also helps with my [other issue](https://github.com/DestinyItemManager/DIM/issues/10222):
1) Allow in-between saving, without closing the whole build popup. Same thing for [save as new](https://github.com/DestinyItemManager/DIM/issues/10222).
2) Add a before leave event handler to prevent refreshes when you have unsaved changes.

**Comment by bhollis** [comment]
We already won't reload-to-update if there's any sheet up:
https://github.com/DestinyItemManager/DIM/blob/master/src/app/dim-ui/AutoRefresh.tsx#L199-L200

However, if one tab upgrades, they all must reload or the upgrade gets stuck.

To fix this we'd need coordination between tabs, which is basically https://github.com/DestinyItemManager/DIM/issues/9733 - something I'd like to do but it's low on the priority list.

I'd suggest using prod DIM instead of beta, since prod generally only updates once a week.

**Comment by Revadike** [comment]
Fair enough, but I think preventing lost changes is more important than upgrading. All you need is a tab refresh anyway, right? Just make it reload after pushing the save button.

**Comment by bhollis** [comment]
"Just" is doing a lot of heavy lifting there, but if you'd like to volunteer a PR...
